pre-commit/app_tests/json_tests.py

```diff
@@ -0,0 +1,457 @@
+            else:
+                message = (
+                    f"GUID and app package name not listed in {APPID_TO_PACKAGE_NAME_FILEPATH}"
+                )
+
+        return create_test_result_response(
+            success=not message, message=TEST_PASS_MESSAGE if not message else message
+        )
+
+    @TestSuite.test
+    def check_action_param_prefixes(self):
+        """
+        Every parameter has an action_result.parameter associated with it
+        """
+        app_json_actions = self._app_json.get("actions")
+
+        message = TEST_PASS_MESSAGE
+        verbose = []
+        for index, action in enumerate(app_json_actions):
+            if action["action"] in ("test connectivity", "on poll"):
+                continue
+            action_parameters = action.get("parameters", {})  # Gets the parameters, if any
+            if not action_parameters:
+                continue
+
+            action_output = self._get_data_paths(action, parameters=True)
+            action_output_path_values = set(action_output.keys())
+
+            for param_name in action_parameters:
+                formatted_param = f"action_result.parameter.{param_name}"
+                if formatted_param not in action_output_path_values:
+                    # if this check fails that means the validate_json_schema test will also fail which is why it's fine to do this
+                    if "data_type" in action_parameters[param_name]:
+                        new_output_param = {
+                            "data_path": formatted_param,
+                            "data_type": action_parameters[param_name]["data_type"],
+                        }
+                        self._app_json["actions"][index]["output"].append(new_output_param)
+                        verbose.append(f"Missing action result output for {formatted_param}")
+
+            parameters_formatted = set(
+                f"action_result.parameter.{name}" for name in action_parameters
+            )
+            action_output_to_remove = []
+            for data_path, idx_param in action_output.items():
+                if data_path not in parameters_formatted:
+                    action_output_to_remove.append(idx_param)
+                    verbose.append(f"Extra action result output for {data_path}")
+
+            for idx_param in reversed(action_output_to_remove):
+                self._app_json["actions"][index]["output"].pop(idx_param)
+
+        if verbose:
+            message = f"Action results should only contain an action_result.parameter key for every action parameter. Modifying {self._parser.app_json_name}"
+            self._parser.update_app_json(self._app_json)
+
+        return create_test_result_response(
+            success=not verbose, message=message, verbose=verbose, fixed=True
+        )
+
+    @TestSuite.test(remove_tags=[DEVELOPER_SUPPORTED])
```

Comment:
If this auto-fixes, we should be able to run it against all connectors
```suggestion
    @TestSuite.test
```